### ISSUE TITLE
Attempting to reconnect a disconnected client produces a zlib incorrect header check

### ISSUE THREAD

**Issue by starcraft66** [ammaraskar/pyCraft#109]
In my code, I have a packet listener that listend for a disconnect packet and attempts to reconnect the client, however if I call `connection.connect()` on this now disconnected connection, I get the following error.
```Exception in thread Networking Thread:
Traceback (most recent call last):
  File "/home/tristan/.local/share/virtualenvs/minecraft-discord-bridge-wee7owyS/lib/python3.6/site-packages/minecraft/networking/connection.py", line 460, in run
    self._run()
  File "/home/tristan/.local/share/virtualenvs/minecraft-discord-bridge-wee7owyS/lib/python3.6/site-packages/minecraft/networking/connection.py", line 494, in _run
    self.connection.file_object, timeout=read_timeout)
  File "/home/tristan/.local/share/virtualenvs/minecraft-discord-bridge-wee7owyS/lib/python3.6/site-packages/minecraft/networking/connection.py", line 548, in read_packet
    packet_data.read())
zlib.error: Error -3 while decompressing data: incorrect header check

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/home/tristan/.local/share/virtualenvs/minecraft-discord-bridge-wee7owyS/lib/python3.6/site-packages/minecraft/networking/connection.py", line 464, in run
    self.connection._handle_exception(e, sys.exc_info())
  File "/home/tristan/.local/share/virtualenvs/minecraft-discord-bridge-wee7owyS/lib/python3.6/site-packages/minecraft/networking/connection.py", line 410, in _handle_exception
    raise_(*exc_info)
  File "/home/tristan/.local/share/virtualenvs/minecraft-discord-bridge-wee7owyS/lib/python3.6/site-packages/future/utils/__init__.py", line 413, in raise_
    raise exc.with_traceback(tb)
  File "/home/tristan/.local/share/virtualenvs/minecraft-discord-bridge-wee7owyS/lib/python3.6/site-packages/minecraft/networking/connection.py", line 460, in run
    self._run()
  File "/home/tristan/.local/share/virtualenvs/minecraft-discord-bridge-wee7owyS/lib/python3.6/site-packages/minecraft/networking/connection.py", line 494, in _run
    self.connection.file_object, timeout=read_timeout)
  File "/home/tristan/.local/share/virtualenvs/minecraft-discord-bridge-wee7owyS/lib/python3.6/site-packages/minecraft/networking/connection.py", line 548, in read_packet
    packet_data.read())
zlib.error: Error -3 while decompressing data: incorrect header check```

**Comment by joodicator** [comment]
Thank you for this report. This is indeed a bug in pyCraft, caused by a failure to reset the connection's `options.connection_enabled` to `False` upon reconnection, for which I am to blame.

This has been fixed in commit 48e1003.

The bug never came to light until now because, I think, the main real-world use of the reconnection ability occurred before compression was enabled, i.e. during version negotiation. In my own use of pyCraft, I just recreated the `Connection` object each time I had to reconnect, because it didn't even nominally support reconnection at that time. 48e1003 also adds a regression test for this bug to make sure reconnection works properly when encryption and compression are enabled.

@starcraft66 I see that you're working on a Minecraft/Discord bridge - interesting! I have used pyCraft to make a Minecraft/IRC bridge before, and I might have use for a Discord bridge in the future. Please let us know if you have any further issues with reconnection.

**Comment by starcraft66** [comment]
Thanks for the quick bug fix @joodicator! I am indeed writing a minecraft to discord bridge, it is currently far from finished and the implementation to-do list is quite large but it will probably start looking quite usable in a week or two :)

I will let you know if I have any further issues when I get to to working on the bridge again tonight!
